scrapers_next/me/people.py

```diff
@@ -0,0 +1,328 @@
+                XPath("//*[@id='content']/p/strong[contains(text(), 'State')]")
+                .match_one(self.root)
+                .tail.strip()
+            )
+            state_phone = state_phone.lstrip(":").strip()
+            p.capitol_office.voice = state_phone
+        except SelectorError:
+            pass
+
+        try:
+            state_phone = (
+                XPath("//*[@id='content']/p/b[contains(text(), 'State')]")
+                .match_one(self.root)
+                .tail.strip()
+            )
+            state_phone = state_phone.lstrip(":").strip()
+            p.capitol_office.voice = state_phone
+        except SelectorError:
+            pass
+
+        website = (
+            XPath("//*[@id='content']/p/strong[contains(text(), 'Website')]")
+            .match_one(self.root)
+            .getnext()
+        )
+        if website.get("href") is None:
+            website = website.getnext().get("href")
+        else:
+            website = website.get("href")
+        p.extras["website"] = website
+
+        return p
+
+
+class Senate(ExcelListPage):
+    source = URL(
+        "https://legislature.maine.gov/uploads/visual_edit/130th-senate-contact-information-as-of-31221.xlsx"
+    )
+
+    def process_item(self, item):
+        # skip header and empty rows
+        if item[0] == "Dist" or item[0] is None:
+            self.skip()
+
+        first_name = item[3].strip()
+        last_name = item[4].strip()
+        name = first_name + " " + last_name
+        district = item[0]
+        party = item[2].strip()
+
+        p = ScrapePerson(
+            name=name,
+            state="me",
+            chamber="upper",
+            district=district,
+            party=party,
+        )
+
+        p.given_name = first_name
+        p.family_name = last_name
+
+        detail_link = URL(f"https://legislature.maine.gov/District-{district}")
+        p.add_source(self.source.url)
+        p.add_source(detail_link.url)
+        p.add_link(detail_link.url, note="homepage")
+
+        county = item[1].strip()
+        p.extras["county represented"] = county
+
+        mailing_address = item[5].strip()
+        city = item[6].strip()
+        zip = item[8].strip()
+        address = mailing_address + ", " + city + ", ME " + zip
+        if re.search(r"St(\s|,)", address):
+            address = re.sub(r"St\s", "Street ", address)
+            address = re.sub(r"St,", "Street,", address)
+        if re.search(r"PO\s", address):
+            address = re.sub(r"PO\s", "P.O. ", address)
+        if re.search(r"Saint\s", address):
+            address = re.sub(r"Saint\s", "St. ", address)
+        if re.search(r"N\.", address):
+            address = re.sub(r"N\.", "North", address)
+        p.extras["Mailing address"] = address
+        # should this be a district office?
+
+        phone = item[9].strip()
+        phone = "(207) " + phone
+        # append area code to phone?
+        p.extras["phone"] = phone
```

Comment:
any reason for these addresses/phone #s  to be on extras?

Reply:
Wasn't sure if these should be assigned to capitol or district offices. I can change to district.